File: back end/security-service/src/main/java/com/cibertec/service/AuthService.java
package com.cibertec.service;

import com.cibertec.client.dto.UserRequest;
import com.cibertec.client.dto.UserResponse;
import com.cibertec.dto.AuthRequest;
import com.cibertec.dto.AuthResponse;
import jakarta.validation.Valid;

public interface AuthService {
    AuthResponse login(AuthRequest authRequest);
    UserResponse register(UserRequest userRequest);
    UserResponse registerAdmin(UserRequest userRequest);
}

File: la-gula-pizzeria/src/app/shared/services/auth/auth-service.ts
import { Injectable } from '@angular/core';
import { GATEWAY_URL, SERVICES_PATHS } from '../config/config';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import AuthResponse from '../../model/auth/AuthResponse.model';
import AuthRequest from '../../model/auth/AuthRequest.model';
import UserRequest from '../../model/user/request/userRequest.model';
import UserResponse from '../../model/user/response/userResponse.model';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  path: string = `${GATEWAY_URL}${SERVICES_PATHS.SECURITY}` + '/auth';

  constructor(private http: HttpClient) {
  }
  login(authRequest: AuthRequest) : Observable<AuthResponse> {
    return this.http.post<AuthResponse>(`${this.path}` + '/login', authRequest);
  }
  register(userRequest: UserRequest) : Observable<UserResponse> {
    return this.http.post<UserResponse>(`${this.path}` + '/register', userRequest);
  }
}
